packages/web3-providers-http/src/index.ts

```diff
@@ -78,9 +79,14 @@ export default class HttpProvider<
 			},
 			body: JSON.stringify(payload),
 		});
-
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-		if (!response.ok) throw new ResponseError(await response.json());
+		if (!response.ok) { 
```

Comment:
Instead of this http provider, can you do this in [quicknode provider](https://github.com/web3/web3.js/blob/7f8e504bae5f4c02b8cf1cc09b41a1c3757ddc95/packages/web3-rpc-providers/src/web3_provider_quicknode.ts#L23), so we add Quicknode specific message ?